src/main/java/com/intellij/compiler/notNullVerification/ImplicitThrowOnNullMethodVisitor.java

```diff
@@ -30,11 +30,23 @@
     ImplicitThrowOnNullMethodVisitor(@Nullable final MethodVisitor methodVisitor, @NotNull final Type[] argumentTypes, @NotNull final Type returnType, final int access, @NotNull final String methodName, @NotNull final String className, @NotNull final Set<String> nullableAnnotations) {
         super(Opcodes.ASM5, methodVisitor, argumentTypes, returnType, access, methodName, className, true);
         this.nullableAnnotations = nullableAnnotations;
-        if (!isSynthetic()) {
+        if (!isSynthetic() && !isAbstractInnerClassConstructor()) {
             addImplicitNotNulls();
         }
     }
 
+    private boolean isAbstractInnerClassConstructor() {
+        return isInnerClass() && isConstructor();
+    }
+
+    private boolean isInnerClass() {
```

Comment:
Fixed that too now, had "Abstract" there also for some reason probably wrote that instead of Anonymous which I was thinking...

Thanks for the feedback!